### ISSUE TITLE
[FEATURE] create `metrics` module

### ISSUE THREAD

**Issue by davidberenstein1957** [recognai/rubrix#4034]
**Problem Description:**
I believe that users are looking for basic metrics related to annotators.

**Solution Proposal:**

*Metrics for Annotator Suggestions vs. Responses*
We should be able to obtain metrics for individual questions or sets of questions, depending on the nature of the questions. The desired annotation metrics include:

- For LabelQuestion and MultiLabelQuestion:
    - F1 score
    - Accuracy
    - Recall
    - Precision
    - Confusion matrix

- For RatingQuestion (considering differences between max_rating and min_rating):
    - There is correlation to the ranks so x+1 is a worse mistake than x+2
    - Comparable to LabelQuestions

- For RankingQuestion (addressing ranking ties):
    - Comparable with LabelQuestions

- For TextQuestion (focusing on text-based metrics):
    - BLEU
    - ROUGE
    - Edit distance (e.g., Levenshtein)
    - LLM-specific metrics may also be considered

*Metrics for Unified Responses and Suggestions*
In addition to annotator-specific metrics, we need to unify metrics based on the criteria defined in the [documentation](https://docs.argilla.io/en/latest/practical_guides/collect_responses.html#unifying-disagreements) to ensure a consistent view of the data. This applies to LabelQuestion, MultiLabelQuestion, RatingQuestion, and RankingQuestion.

For TextQuestion, we cannot directly unify written texts. However, responses can be unified using a disagreement approach, averaging the responses during metric computation.

**Code Implementation:**
The proposed code snippet for achieving this functionality:

```python
import argilla as rg

rg.FeedbackDataset.unify_responses("<question_name>", "<metric_name>")
```

**Inter-Annotator Agreement:**
To gather information on inter-annotator agreements, we can refer to the metrics used in the Toloka's crowd-kit for various question types:

- LabelQuestion and RatingQuestion: [Categorical Responses](https://github.com/Toloka/crowd-kit#categorical-responses)
- MultiLabelQuestion: [Multi-Label Responses](https://github.com/Toloka/crowd-kit#multi-label-responses)
- RankingQuestion: [Pairwise Comparisons](https://github.com/Toloka/crowd-kit#pairwise-comparisons)
- TextQuestion: [Textual Responses](https://github.com/Toloka/crowd-kit#textual-responses)

These concepts can be integrated into the FeedbackDataset class using the following code:

```python
import argilla as rg

rg.FeedbackDataset.compute_metrics("<question_name>", "<metric_name>")
```

Alternatively, the metrics can be implemented as a separate module:

```python
from argilla.feedback.metrics import Metrics

Metrics(ds, "<question_name>", "<metric_name>")
```

**Alternatives Considered:**
None mentioned.

**Additional Context:**
No additional context provided.

**Comment by davidberenstein1957** [comment]
https://github.com/argilla-io/argilla/issues/2134

**Comment by davidberenstein1957** [comment]
@dvsrepo did you have anything to add here?

**Comment by dvsrepo** [comment]
Nothing else than the summary we prepared on Friday

**Comment by plaguss** [comment]
Hi @frascuchon I was thinking on adding a separate module for the metrics as suggested by @davidberenstein1957 but it seems there is already an `argilla.metrics` module. Does it make sense to include these metrics there, or should we put them in a different place?

**Comment by cceyda** [comment]
Could possibly use huggingface [evaluate](https://github.com/huggingface/evaluate) on the metric calculation side and just interface with that 🤗 
I would love a good UI visual integration for stuff like; confusion matrix (click on the matrix to filter down to those stuff), edit distance (heatmap like view) etc 🤔

**Comment by davidberenstein1957** [comment]
Hi @cceyda, we are hard at work on the `metrics` module but were hoping to limit additional external dependencies, like `evaluate`.  Feel free to have a look at this [active PR here](https://github.com/argilla-io/argilla/pull/4271), feedback is very welcome :)

**Comment by davidberenstein1957** [comment]
W.r.t. the UI and visual integrations, we expect to allow for some basic support for this but generally, these more advanced functionalities that are aimed at power users and larger scale companies will be part of Argilla cloud.

**Comment by plaguss** [comment]
Closed per #4271
